http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -1248,6 +1249,12 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
         if (req.appType == AppType.Allowed)
           Some(config.leoKubernetesConfig.allowedAppConfig.numOfReplicas)
         else None
+      autoDeleteThresholdInMinutes = req.autoDeleteThresholdDuration match {
+        case None               => autoDeleteOffValue
+        case Some(Duration.Inf) => autoDeleteOffValue
```

Comment:
Continuing the conversation from [the other PR](https://github.com/DataBiosphere/leonardo/pull/3943#discussion_r1417385887) here.

Reply:
wrote the thinking down in [here](https://docs.google.com/document/d/1eWfY8PQoDKVOdIeREuEC9ObLhXlsTtG2ugLwuqov81k/edit#heading=h.2b36mygo7cbv)